responsemanager/queryexecutor/queryexecutor.go

```diff
@@ -195,18 +196,28 @@ func (qe *QueryExecutor) runTraversal(p peer.ID, taskData ResponseTask) error {
 			}
 			return err
 		}
-		lnk, data, err := qe.nextBlock(taskData)
+		lnk, _ := taskData.Traverser.CurrentRequest()
+		ctx, span := otel.Tracer("graphsync").Start(ctx, "processBlock", trace.WithAttributes(
+			attribute.String("cid", lnk.String()),
+		))
+		lnk, data, err := qe.nextBlock(ctx, taskData)
 		if err != nil {
+			span.End()
 			return err
 		}
-		err = qe.sendResponse(p, taskData, lnk, data)
+		err = qe.sendResponse(ctx, p, taskData, lnk, data)
 		if err != nil {
+			span.End()
 			return err
 		}
+		span.End()
 	}
 }
 
-func (qe *QueryExecutor) nextBlock(taskData ResponseTask) (ipld.Link, []byte, error) {
+func (qe *QueryExecutor) nextBlock(ctx context.Context, taskData ResponseTask) (ipld.Link, []byte, error) {
```

Comment:
maybe this should be renamed now to `loadBlock`, or call it `loadCurrentBlock` and don't return the link since it's now being pulled up to in the caller for span metadata so it's a pointless variable overwrite.